mlir/utils/performance/tuningRunner.py

```diff
@@ -165,76 +191,166 @@ def get_winning_config(tuning_output, test_vector, config, all_data, paths: Path
+        ]
+
+        for test_vector in configs:
+            if not test_vector.endswith(".mlir"):
+                command_line = test_vector.split(sep=' ')
+                config = conf_class.from_command_line(command_line, options.arch, options.num_cu)
+                test_vector = config.to_command_line()
+                print("Tuning:", test_vector, file=sys.stderr)
+                command_line_options = config.generate_mlir_driver_commandline(
+                    options.rocmlir_gen_flags, kernel_repeats=MLIR_N_REPEATS)
+                # Note, we don't need the -ph, this goes to the tuning driver
+                kernel_gen_command = paths.mlir_paths.rocmlir_gen_path + ' ' + command_line_options
+                kernel_gen = subprocess.Popen(kernel_gen_command.split(),
+                                              stdout=subprocess.PIPE,
+                                              stderr=subprocess.DEVNULL)
+                tuning_loop = subprocess.Popen([paths.mlir_paths.rocmlir_tuning_driver_path] +
+                                               tuning_driver_args,
+                                               stdin=kernel_gen.stdout,
+                                               stdout=subprocess.PIPE,
+                                               stderr=subprocess.PIPE)
+                kernel_gen.stdout.close()
+            else:
+                # pipe to rocmlir_gen --emit-tuning-key
+                tuning_key = subprocess.Popen(
+                    [paths.mlir_paths.rocmlir_gen_path, '--emit-tuning-key', test_vector],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE)
+                output, _ = tuning_key.communicate()
+                result = output.decode('utf-8').strip().split('\t')
+                print(f"Tuning:{result[2]} from {test_vector}", file=sys.stderr)
+                command_line = result[2].split(sep=' ')
+                config = conf_class.from_command_line(command_line, options.arch, options.num_cu)
+                tuning_loop = subprocess.Popen([paths.mlir_paths.rocmlir_tuning_driver_path] +
+                                               tuning_driver_args + [test_vector],
+                                               stdout=subprocess.PIPE,
+                                               stderr=subprocess.PIPE)
+
+            error_title = f"Error tuning test vector: {test_vector}"
+            try:
+                # Tune, printing progress as we go to avoid CI timeouts
+                winning_config, max_tflops, entries = get_winning_config(
+                    tuning_loop.stdout, test_vector, config, paths, options)
+            except TuningError as e:
+                log_error(error_title, str(e), outfile)
+                if options.abort_on_error:
+                    return False
+                else:
+                    continue
+
+            tuning_loop.wait()
```

Comment:
If exception occured then this it would never reach wait and can cause zombie processes.